course_catalog/tasks.py

```diff
@@ -34,3 +40,48 @@ def get_edx_data():
             break
 
         url = response.json()["next"]
+
+
+@task
+def get_ocw_data():
+    """
+    Task to sync OCW course data with database
+    """
+    raw_data_bucket = boto3.resource(
+        's3',
+        aws_access_key_id=settings.OCW_CONTENT_ACCESS_KEY,
+        aws_secret_access_key=settings.OCW_CONTENT_SECRET_ACCESS_KEY
+    ).Bucket(name=settings.OCW_CONTENT_BUCKET_NAME)
+
+    # Get all the courses keys
+    ocw_courses = set()
+    # print("Assembling list of courses...")
+    for file in raw_data_bucket.objects.all():
+        key_pieces = file.key.split("/")
+        course_prefix = key_pieces[0] + "/" + key_pieces[1]
+        if course_prefix not in NON_COURSE_DIRECTORIES:
+            if "/".join(key_pieces[:-2]) != "":
+                ocw_courses.add("/".join(key_pieces[:-2]))
+
+    # Query S3 Bucket for JSONs per course
+    for course_prefix in ocw_courses:
+        loaded_raw_jsons_for_course = []
+        last_modified_dates = []
+        # print("Digesting: " + course_prefix + " ...")
+        for obj in raw_data_bucket.objects.filter(Prefix=course_prefix):
+            loaded_raw_jsons_for_course.append(load_json_from_string(obj.get()["Body"].read()))
+            last_modified_dates.append(obj.get()["LastModified"])
+
+        parser = OCWParser("", "", loaded_raw_jsons_for_course)
```

Comment:
Can you start the `try` block just above here?  When running it I got an exception at this point, which prevented susbsequent courses from getting processed.

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-2641f00b6d13> in <module>
----> 1 get_ocw_data()

/usr/local/lib/python3.6/site-packages/celery/local.py in __call__(self, *a, **kw)
    189
    190     def __call__(self, *a, **kw):
--> 191         return self._get_current_object()(*a, **kw)
    192
    193     def __len__(self):

/usr/local/lib/python3.6/site-packages/celery/app/task.py in __call__(self, *args, **kwargs)
    373         self.push_request(args=args, kwargs=kwargs)
    374         try:
--> 375             return self.run(*args, **kwargs)
    376         finally:
    377             self.pop_request()

/src/course_catalog/tasks.py in get_ocw_data()
     73             last_modified_dates.append(obj.get()["LastModified"])
     74
---> 75         parser = OCWParser("", "", loaded_raw_jsons_for_course)
     76         # parser.setup_s3_uploading(settings.OCW_LEARNING_COURSE_BUCKET_NAME,
     77         #                           settings.OCW_LEARNING_COURSE_ACCESS_KEY,

/usr/local/lib/python3.6/site-packages/ocw_data_parser/ocw_data_parser.py in __init__(self, course_dir, destination_dir, loaded_jsons, upload_to_s3, s3_bucket_name, s3_bucket_access_key, s3_bucket_secret_access_key, s3_target_folder)
     54             self.jsons = loaded_jsons
     55         if self.jsons:
---> 56             self.master_json = self.generate_master_json()
     57             self.destination_dir += safe_get(self.jsons[0], "id") + "/"
     58

/usr/local/lib/python3.6/site-packages/ocw_data_parser/ocw_data_parser.py in generate_master_json(self)
    131         new_json["url"] = safe_get(self.jsons[0], "technical_location").split("ocw.mit.edu")[1]
    132         new_json["short_url"] = safe_get(self.jsons[0], "id")
--> 133         new_json["effective_date"] = _format_date(safe_get(self.jsons[0], "effectiveDate"))
    134         new_json["expiration_date"] = _format_date(safe_get(self.jsons[0], "expirationDate"))
    135         new_json["image_src"] = self.course_image_s3_link

/usr/local/lib/python3.6/site-packages/ocw_data_parser/ocw_data_parser.py in _format_date(date_str)
    105                 tz = date_pieces.pop(2)
    106                 timezone = pytz.timezone(tz) if "GMT" not in tz else pytz.timezone("Etc/" + tz)
--> 107                 tz_stripped_date = datetime.strptime(" ".join(date_pieces), "%Y-%m-%d %H:%M:%S")
    108                 tz_aware_date = timezone.localize(tz_stripped_date)
    109                 tz_aware_date = tz_aware_date.astimezone(pytz.utc)

/usr/local/lib/python3.6/_strptime.py in _strptime_datetime(cls, data_string, format)
    563     """Return a class cls instance based on the input string and the
    564     format string."""
--> 565     tt, fraction = _strptime(data_string, format)
    566     tzname, gmtoff = tt[-2:]
    567     args = tt[:6] + (fraction,)

/usr/local/lib/python3.6/_strptime.py in _strptime(data_string, format)
    363     if len(data_string) != found.end():
    364         raise ValueError("unconverted data remains: %s" %
--> 365                           data_string[found.end():])
    366
    367     iso_year = year = None

ValueError: unconverted data remains: .607
```

Reply:
I have ran into this error few times while syncing OCW courses.  I will remove `effective_date` and `expiration_date` from `ocw-data-parser` since we are leaving `start_date` and `end_date` blank/null for OCW courses

Reply:
I can't remember if we want `effective_date` and `expiration_date` fields or not, but it would be good to have this inside the try block anyway in case some other kind of error pops up.

Reply:
@pdpinch mentioned in yesterday's standup that we will not need those fields that's why I am leaning towards removing them.

Reply:
I'm confirming that we don't need `start_date` and `end_date` for the OCW courses. 

What exactly is the problem with `effective_date` and `expiration_date`? Can you give me an example of good and bad values?